### DISCUSSION TITLE
Grafonnet

### DISCUSSION THREAD

**Discussion by ebattat** [community/community#57325] [Programming Help]
### Select Topic Area

Question

### Body

Hi,

Is there any auto completion plugin that I can use for [grafonnet](https://github.com/grafana/grafonnet) ?

**Comment by Abbas-Aljuaifari** [comment]
there isn't a specific auto-completion plugin exclusively designed for Grafonnet. However, Grafonnet is a library for generating Grafana dashboards using JSONnet, which means that you can leverage the capabilities of JSONnet-aware code editors or integrated development environments (IDEs) to enhance your development experience.

Popular code editors like Visual Studio Code (VS Code) have extensions that support JSONnet, which can provide auto-completion and other helpful features. One such extension is called "vscode-jsonnet," which adds JSONnet support to VS Code. This extension includes features like syntax highlighting, auto-completion, code formatting, and more.

To install the "vscode-jsonnet" extension in Visual Studio Code, you can follow these steps:

Open Visual Studio Code.
Navigate to the Extensions view by clicking on the square icon on the left sidebar or by pressing Ctrl+Shift+X (Cmd+Shift+X on macOS).
Search for "vscode-jsonnet" in the extensions marketplace.
Click on the extension offered by 'Dominik Tornow' and click the "Install" button.
Once you have the extension installed, it should provide auto-completion and other useful features when editing Grafonnet files within Visual Studio Code.
